config/routes.rb

```diff
@@ -2,6 +2,7 @@
   devise_for :users
   root "home#index"
   get 'profile', to: 'home#profile'
-  get 'setting', to: 'home#setting'
+  get 'setting', to: 'setting#show'
   post 'fetchnews', to: 'home#fetch_news'
+  post 'setting/update', to: 'setting#update'
```

Comment:
and for updating you should use PUT request (you can google difference between PUT and POST. It is really useful)